### ISSUE TITLE
Python shell on CogServer doesn't handle multi-line editing needed for functions

### ISSUE THREAD

**Issue by inflector** [opencog/opencog#1338]
You can't currently do this:

```
py>def some_function():
...    print "hello"
...
py>
```

instead you get an infinite set of:

```
py>def some_function():
...    print "hello"
...
...
...
...
```

The Python shell handling code should process the entire set of multi-lines when it gets a blank line following the `...`. That's what Python does if you start it directly from the command line.

**Comment by keyvan-m-sadeghi** [comment]
>
> You can't currently do this:
>
> py>def some_function():
> ...print "hello"
> ...
> py>
>
> Do you have the indents spaced correctly? Should be more like:

py>def some_function():
...    print "hello"
...
py>


not sure, that used to work two years ago...

**Comment by inflector** [comment]
Yes, sorry, the indents were fine in the code I was trying. I'll edit the original report above to reflect that.

**Comment by inflector** [comment]
And Python should report an error in any case if the indents are wrong.

**Comment by inflector** [comment]
Turns out GenericShell::do_eval was eating the empty string case before it got to the code that was adding a '\n' because the telnet socket code was eating newlines.

**Comment by linas** [comment]
put newlines back in, earlier. Its harmless.

**Comment by inflector** [comment]
I just removed the:

```
if (0 == len)
	{
		put_output(get_prompt());
		return;
	}
```

at the top, and put the special character handling code into an:

```
if  (len > 0)
	{
		SPECIAL CHARACTER CODE
	}
```

so the code skips it in the empty string case and drops down to the section that adds the `'\n'`.

**Comment by inflector** [comment]
Fixed in pull request #1348
